snippets/csharp/keywords/GenericWhereConstraints.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+using static keywords.UnmanagedExtensions;
+
+namespace keywords
+{
+    // <Snippet1>
+    public class AGenericClass<T> where T : IComparable { }
+    // </Snippet1
+
+    // <SNippet2>
+    public class UsingEnum<T> where T : System.Enum { }
+
+    public class UsingDelegate<T> where T : System.Delegate { }
+
+    public class Multicaster<T> where T : System.MulticastDelegate { }
+    // </Snippet2>
+
+    // <Snippet3>
+    class MyClass<T, U>
+        where T : class
+        where U : struct
+    { }
+    // </Snippet3>
+
+    // <Snippet4>
+    class UnManagedWrapper<T>
+        where T : unmanaged
+    { }
+    // </Snippet4>
+
+    // <Snippet5>
+    public class MyGenericClass<T> where T : IComparable, new()
+    {
+        // The following line is not possible without new() constraint:
+        T item = new T();
+    }
+    // </Snippet5>
+
+    // <Snippet6>
+    public interface IMyInterface
+    {
+    }
```

Comment:
nit: for the sake of the screen space, I would put all the declaration in one line.